File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row=len(matrix)
        col=len(matrix[0])
        low=0
        high=(row*col)-1
        while low<=high:
            mid=(low+high)//2
            row=mid//col
            cols=mid%col
            if matrix[row][cols]==target:
                return True
            elif matrix[row][cols]<target:
                low=mid+1
            else:
                high=mid-1
        return False
        

File: Data Structures & Algorithms/find-first-and-last-position-of-element-in-sorted-array/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target){
        int low = 0,high=nums.size()-1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(nums[mid]>=target) high = mid - 1;
            else low = mid + 1;
        }
        return low;
    }
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> res;
        if ((int)nums.size() == 0) return {-1,-1};
        int a = search(nums,target);
        int b = search(nums,target+1) - 1;
        if(nums[a]!=target) return {-1,-1};
        res.push_back(a);
        res.push_back(b);
        return res;
    }
};